cloud/scope/machine.go

```diff
@@ -299,6 +299,15 @@ func (m *MachineScope) ProviderID() string {
 	return parsed.ID()
 }
 
+// AvailabilitySet returns the availability set for this machine if available
+func (m *MachineScope) AvailabilitySet() (string, bool) {
+	if len(m.ClusterScoper.FailureDomains()) == 0 && m.IsControlPlane() {
```

Comment:
Just curious, have you thought about this approach can be expanded later to add availability sets for worker nodes part of machine deployments?

Reply:
I did, it shouldn't be too hard, if we just have one AS for all worker node (of all machine deployments). That's why I've left availability spec as an array, so that if it'd be easier to accommodate the changes, if needed. Happy to do it in this pr, if everyone's okay with the approach.

Reply:
let's do worker nodes in a separate PR to keep the changes well-scoped and easier to review